feature-libs/quote/root/guards/quote-cart.guard.ts

```diff
@@ -31,19 +37,33 @@ export class QuoteCartGuard implements CanActivate {
           routerState,
           isCheckoutAllowed
         );
-
         if (isQuoteCartActive && !isAllowedCheckoutNavigation) {
-          this.routingService.go({
-            cxRoute: 'quoteDetails',
-            params: { quoteId: quoteId },
-          });
-          return false;
+          const pathForRoute = this.validateThatPresent(
+            'Route `quoteDetails` must be present',
+            this.semanticPathService.get('quoteDetails')
+          );
+          const path = pathForRoute.replace(
+            ':quoteId',
+            this.validateThatPresent(
+              'QuoteId must be present in case cart is a quote cart',
+              quoteId
+            )
+          );
+          console.log('CHHI path: ' + path);
+          return this.router.parseUrl(path);
```

Comment:
[MEDIUM]
Instead of manually replacing a sub-string with `:paramName` in the returned configured string, what about using `semanticPathService.transform()` which can already consume route name and dynamic params, so produces ?

```ts
return this.router.createUrlTree(
  this.semanticPathService.transform({ path: 'quoteDetails', params: { quoteId }})
)
```